### ISSUE TITLE
.git is not a group

### ISSUE THREAD

**Issue by rmoehn** [clojure-grimoire/lib-grimoire#20]
When I run `list-namespaces` on a freshly cloned https://github.com/clojure-grimoire/doc-clojure-core, I get this:

```clojure
(#g/t [:grimoire.things/group {:name ".git", :grimoire.things/url ".git"}] #g/t [:grimoire.things/group {:name "org.clojure", :grimoire.things/url "org.clojure"}])
```

Slightly surprising. The reason is that `.git` is a directory and the listing procedures from [grimore.api.fs.read](https://github.com/clojure-grimoire/lib-grimoire/blob/master/src/grimoire/api/fs/read.clj) just return every directory they can find.

Why didn't this occur to you before? I thought Grimoire also uses the above repository.

Anyway, will you fix this yourself or should I? How would you like it fixed? Only list directories that contain a `meta.edn`? Or only list directories that are not hidden? Should I add the same code in all listing methods? Or factor out a procedure that checks both for directory and what we want to check for?
Your style is quite different from mine, so I'm rather unsure how to do it.

**Comment by arrdem** [comment]
So if you look at [how grimoire manages resources](https://github.com/clojure-grimoire/grimoire/blob/master/dat.sh), you'll see that none of those data repositories gets used "in the raw" as you are trying to do. I maintain a "data" directory, being `dat/` into which the various data repos get cloned, and then two parallel directories `notes-store/` and `doc-store/` which have symlinks to the appropriate entries in the data repos. This is done so that the documentation artifacts for artifacts in groups are decomplected rather than requiring monorepos for a group or for all groups. The idea is somewhat along the lines of what you are working on, that these repos could be vendored by a documentation source and I don't have to know what they are or how they are licensed.

TL;DR You may not be using it wrong, but you aren't using it the way I do. Whether this is a bug or not is up for grabs.

**Comment by rmoehn** [comment]
I'd say the library you provide should work out of the box with the data you provide. Right now it's a bit annoying. (Although easy to work around.)

**Comment by arrdem** [comment]
So I think this is actually a bug unless your .git repo has a file `.git/meta.edn`, because having that meta file is what distinguishes a "child" from some random directory.

**Comment by rmoehn** [comment]
Not that I know of.

**Comment by arrdem** [comment]
Okay yeah bug then. the `fs` implementation of `list-groups` should be checking for a meta.edn file...
